tests/std/include/range_algorithm_support.hpp

```diff
@@ -432,7 +452,8 @@ namespace test {
         // iterator unwrapping operations:
         using _Prevent_inheriting_unwrap = iterator;
 
-        using unwrap = iterator<Category, Element, Diff, Eq, Proxy, IsWrapped::no>;
+        using unwrap = std::conditional_t<derived_from<Category, contiguous>, Element*,
+            iterator<Category, Element, Diff, Eq, Proxy, IsWrapped::no>>;

```

Comment:
This appears to be causing test failures:
```
C:\agent\_work\1\s\tests\std\tests\P0896R4_ranges_test_machinery\test.cpp(47): error C2338: same_as<_Unwrapped_t<I>, iterator<Category, Element, Diff, Eq, Proxy, IsWrapped::no>>
C:\agent\_work\1\s\tests\std\tests\P0896R4_ranges_test_machinery\test.cpp(139): note: see reference to function template instantiation 'bool iter_test<std::contiguous_iterator_tag,int,test::CanDifference::no,test::CanCompare::yes,test::ProxyRef::no,test::IsWrapped::yes>(void)' being compiled
```

Reply:
Yes. That's easily fixed - but the ICEs in the heap algorithm test are more challenging. It's the same "`constexpr` reference-to-pointers" bug that plagues many of the permuting algorithms, I'm narrowing down exactly which bits to snip out.

Reply:
I've added a comment to Microsoft-internal VSO-938163 asking for this bug to be considered equal priority with C++20 Core Language feature work, since `std::ranges` is C++20.

Reply:
Workaround (actually avoidance) applied. The FE team has told me that they are working on VSO-938163 now in hopes of getting it fixed for 16.8. I'll move this PR to Initial Review and un-Draft it once the tests pass.